homeassistant/components/fritzbox/climate.py

```diff
@@ -138,6 +138,12 @@ def target_temperature(self) -> float:
 
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
+        if self.data.lock:
```

Comment:
Both `async_set_preset_mode` and `async_set_hvac_mode` call this. Both check for summer and holiday mode, but only `async_set_hvac_mode` also checks for lock. For `async_set_preset_mode` this check would happen here. 

I would move all this checks here, as @mib1185 suggested.

Reply:
with #142863 we get two places were we should check for a lock.
So maybe we could create just a small check helper (_which checks if in summer/vacation mode or lock is enabled_) and place it in both these functions

Reply:
Hi I have added a helper function within climate, that is intended to be used in both async_set_hvac_mode and async_set_temperature, the test fails otherwise when the functions are called but the setting is not changed

Reply:
I don't see any new commit, do you have this helper only on your local dev environment?

Reply:
We nee dthe helper only to be in `async_set_temperature()` as `async_set_hvac` and `async_set_preset` ends in `async_set_temperature()`

Reply:
Yes, but then the test for test_holiday_summer_mode fails, for example. I couldn't find a good way to fix this test. I think the problem is that the test changes SERVICE_SET_HVAC_MODE and SERVICE_SET_PRESET_MODE, which no longer cause an error. The obvious solution would be to use SERVICE_SET_TEMPERATURE, but when I try that, I get the error service_not_supported. I assume this is related to the fact that holiday or summer mode is active.

Reply:
> We need the helper only to be in `async_set_temperature()` as `async_set_hvac` and `async_set_preset` ends in `async_set_temperature()`

Yes, but then the test for test_holiday_summer_mode fails, for example. I couldn't find a good way to fix this test. I think the problem is that the test changes SERVICE_SET_HVAC_MODE and SERVICE_SET_PRESET_MODE, which no longer cause an error. The obvious solution would be to use SERVICE_SET_TEMPERATURE, but when I try that, I get the error service_not_supported. I assume this is related to the fact that holiday or summer mode is active.

Reply:
@mib1185 @flabbamann what's now the plan to fix that issue?

Reply:
for now ... the conflict needs to be resolved 😬 
We still need a proper PR description in the initial post, after that I think we're to go with this 👍